src/RequestService/RequestService.ts

```diff
@@ -0,0 +1,98 @@
+
+enum METHODS  {
+    GET = 'GET',
+    POST = 'POST',
+    PUT = 'PUT',
+    DELETE = 'DELETE',
+    PATCH = 'PATCH'
+};
+
+
+function queryStringify(data: string) {
+    let params: string[] = [];
+    
+    Object.entries(data).forEach(([key, value]: [string, string]) => {
+        params.push(`${key}=${value}`);
+    })
+
+    return `?${params.join('&')}`
+}
+
+type Options = {
+    data: any,
+    method: string,
+    headers: string,
+    timeout: number
+}
+
+class HTTPTransport {
+    get = (url: string, options: Options) => {
+        if(options.data) {
+            url += queryStringify(options.data);
+        }
+
+        return this.request(url, {data: options.data, headers: options.headers, method: METHODS.GET, timeout:options.timeout});
+    };
+
+    post = (url: string, options: Options) => {
+
+        return this.request(url, {data: options.data, headers: options.headers, method: METHODS.GET, timeout: options.timeout});
+    };
+
+    put = (url: string, options: Options) => {
+
+        return this.request(url, {data: options.data, headers: options.headers, method: METHODS.GET, timeout:options.timeout});
+    };
+    delete = (url : string, options:Options) => {
+
+        return this.request(url, {data: options.data, headers: options.headers, method: METHODS.GET, timeout:options.timeout});
+    };
+
+    request = (url: string, options: Options) => {
+        const {data, method, headers, timeout} = options;
+
+        return new Promise((resolve, reject) => {
+            const xhr = new XMLHttpRequest();
+
+            xhr.open(method, url);
+
+            xhr.onload = function() {
+                return resolve(xhr)
+            }
+
+            xhr.onerror = reject;
+            xhr.onabort = reject;
+            xhr.withCredentials = true;
+            xhr.ontimeout = reject;
+            xhr.timeout = timeout;
+            xhr.setRequestHeader(headers && "Content-type", "application/json")
```

Comment:
Это выглядит довольно странно, кажется, что тут может быть ошибка.
1) в headers должен лежать объект ключ-значение, а не строка. 
2) && булевый оператор, результатом которого будет левый операнд
3) в названии заголовка Content-Type Type должен быть с большой буквы
<img width="294" alt="image" src="https://user-images.githubusercontent.com/37894659/189372667-ee65047d-0940-49ad-a123-75a38edff8f0.png">

Я бы предложил написать вот так
```suggestion
      Object.entries(headers)
        .forEach([key, value] => xhr.setRequestHeader(key, value))
            xhr.setRequestHeader('Content-Type', 'application/json')
```